### PR TITLE
feat: enable sp1 proof generation

### PR DESCRIPTION
**Pull request by chokobole**
Description

This PR enables SP1 proof generation. All `sp1-recursion-program` unit tests have been passed, and a Fibonacci e2e test was also successful. However, the Fibonacci test currently runs ~1.3x slower. See [fibonacci.log](https://github.com/user-attachments/files/17145487/fibonacci.log)

Key Additions

- Implemented `Hintable` traits for `DuplexChallenger` and `FriProof`.
- Implemented `Serialize` and `Deserialize` traits for `FriProof` and `ProverData`.

### CODE REVIEW

**Comment by TomTaehoonKim** [comment]
[bde9394](https://github.com/kroma-network/tachyon/pull/547/commits/bde939457c4709e674fd68c00cd93307715f3fe0): Can you fix sp -> sp1?

File: tachyon/rs/base/container_util.h [hunk 1]
```diff
@@ -45,6 +45,17 @@ absl::Span<R> ConvertRustSliceToCppSpan(rust::Slice<T> slice) {
   return {reinterpret_cast<R*>(slice.data()), slice.size()};
 }
 
+template <typename T>
+rust::Vec<T> CreateEmptyVector(size_t size) {
```
Comment (dongchangYoo) [inline tachyon/rs/base/container_util.h:49]: How about changing the function name to `CreateDefaultVector` or `CreateInitializedVector `? That's why, the returned data might be mistakenly interpreted as a vector of length 0.

**Review by dongchangYoo — CHANGES_REQUESTED** [review]

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Comment by TomTaehoonKim** [comment]
> 2401e7f

@chokobole [2401e7f](https://github.com/kroma-network/tachyon/pull/547/commits/2401e7f0a379adc7ed515be25067d327fef92c29) Can you fix sp -> sp1?

**Review by batzor — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Pull request merged**